### ISSUE TITLE
Remote image not being downloaded/rendered in ImageTexture node

### ISSUE THREAD

**Issue by fcsteagu** [gpac/gpac#591]
I am using an **ImageTexture** whose URL is pointing to a **remote image**. It seems that the image is not downloaded/rendered. **It works with local resources**.

Note: the test was performed on Android.

Please find a very simple example below:
```
              Shape {
						appearance Appearance {
							texture ImageTexture {
								url ["http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=256x256&maptype=satellite&sensor=true"]
							}
							material Material2D {
								emissiveColor 	0 0 0
								filled 			true
								lineProps LineProperties {
									width 0
								}
							}
						}
						geometry Rectangle {
							size 250 250
						} 
					}
```
I have tried with other links but I get the same problem.

This is what I get in the logs:

> I GPAC_WRAPPER: Unknown Message 88
> I GF_LOG_CODEC: [PrivateDec] Codec GPAC Context Loader Processing at 1
> I GF_LOG_MEDIA: [ODM1] Channel 65534 is end of stream
> I Osmo4Renderer: Surface changed from thread Thread[GLThread 598,5,main]
> I LibrariesLoader: Resizing to 1080x1776
> V WRAPPER_JNI: resize::start
> V GPAC_WRAPPER: resize start
> V GPAC_WRAPPER: resize end
> V WRAPPER_JNI: resize::end
> I GF_LOG_MEDIA: [Scene] Inserting new MediaObject 9de269c0 for resource http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=25
> true
> I GF_LOG_MEDIA: [Terminal] Object redirection to http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=256x256&maptype=satellite
> 
> I GF_LOG_MEDIA: [Terminal] Looking for plugin for URL http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=256x256&maptype=sate
> 
> V PhoneStatusBar: setLightsOn(true)
> I GF_LOG_MEDIA: [Terminal] Mime type found: image/png
> I GF_LOG_DASH: /home/traian/work/gpac/gpac_0.6.2_v5/build/android/jni/libgpac/../../../../src/terminal/network_service.c:918 FOUND matching module GPAC Ima
> 
> I GF_LOG_MEDIA: [Terminal] Found input plugin GPAC Image Reader for URL http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=25
> true (image/png)
> I GPAC_WRAPPER: Rebuilding methods for thread -1651771088
> I GPAC_WRAPPER: Getting current Thread -1651771088...
> I GPAC_WRAPPER: Getting current Thread DONE = 0xaee1b460, now registering before exit...
> I GPAC_WRAPPER: Registering DONE for -1651771088
> I GF_LOG_MEDIA: [Terminal] Service connection ACK received from http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=256x256&ma
> o Error
> I GF_LOG_MEDIA: [ODM] Setting up root object for http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=256x256&maptype=satellite
> 
> I GF_LOG_MEDIA: [Terminal] Setting up object streams
> I GF_LOG_MEDIA: [ODM] Creating codec for stream 1
> I GF_LOG_CODEC: [Codec] Found decoder FFMPEG decoder for stream type PNG Image
> I GF_LOG_MEDIA: [Terminal] Registering codec FFMPEG decoder
> I GF_LOG_MEDIA: [Terminal] Channel connection ACK received from http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=256x256&ma
> o Error
> I GF_LOG_MEDIA: [Terminal] Channel 1 connected - 1 objects opened
> I GF_LOG_CODEC: [Codec] Attaching stream 1 to codec FFMPEG decoder
> I GF_LOG_CODEC: [ODM] Creating composition buffer for codec FFMPEG decoder - 1 units 0 bytes each
> I GF_LOG_MEDIA: [ODM1050] No CU available
> I GF_LOG_MEDIA: [Scene] Setup object manager 1050 (MO 0x9de269c0)
> I GF_LOG_SYNC: [SyncLayer] Starting ES 1
> I GF_LOG_MEDIA: [ODM1050] CH1: At OTB 0 starting channel
> W art     : Native thread exiting without having called DetachCurrentThread (maybe it's going to use a pthread_key_create destructor?): Thread[18,tid=12733
> =0x130340a0,"Thread-602"]
> I GF_LOG_MEDIA: [ODM1050 http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=256x256&maptype=satellite&sensor=true] CH1: At OT
>  (clock init 0) - speed 1
> I GF_LOG_SYNC: [SyncLayer] CB status changed - ODM1050: buffering on at OTB 0 (STB 13506) (nb wait on clock: 1)
> I GF_LOG_SYNC: [SyncLayer] ES1: Forcing clock initialization at STB 13507 - AU DTS 0 CTS 0
> I GF_LOG_SYNC: [SyncLayer] ES1 (http://maps.google.com/maps/api/staticmap?center=48.626669,2.433912&zoom=19&size=256x256&maptype=satellite&sensor=true) at
> S 0 - size 45287 - RAP 0
> I GPAC_WRAPPER: Destroying a thread with attached data, env=0xaee33150.
> I GF_LOG_CODEC: [FFMPEG decoder] ODM1050 ES1 at 0 decoding frame DTS 0 CTS 0 size 45287 (0 in channels)
> I GF_LOG_SYNC: [SyncLayer] CB Buffering done ODM1050: buffering off at OTB 0 (STB 13513) (nb wait on clock: 0)
> I GF_LOG_CODEC: [FFMPEG decoder] ODM1050 ES1 At 0 (frame TS 0 - 4442 us ): decoded error Feature Not Supported
> I GF_LOG_MEDIA: [ODM1050] Channel 1 is end of stream
> I GF_LOG_CODEC: [FFMPEG decoder] ODM1050 reordering mode - first frame dispatch - CTS 0 - min TS diff 0
> I GF_LOG_CODEC: [FFMPEG decoder] Exit decode loop because no more input data
> I GF_LOG_MEDIA: [Compositor] done flushing frame in 0 ms
> I GF_LOG_MEDIA: [ODM1050] Switching composition memory to stop state - time 0
> I GPAC_WRAPPER: EOS Reached (103)
> I GF_LOG_MEDIA: [ODM1050] No CU available
> I GF_LOG_MEDIA: [Compositor] done flushing frame in 0 ms

**Comment by fcsteagu** [comment]
hmm it seems that it works in the last nightly build..
I'll keep the post updated if I encounter the same issue in the future.

**Comment by fcsteagu** [comment]
Actually the problem comes from the image decoder library. On Android 5 I must manually set codec_04_6D to FFMPEG decoder. If I leave it like this, the PNG images are not decoded on Android 6. I tried to open PNG images on a device running Android 5.1.1 and the image is not decoded/rendered. I used the last nightly build of the player.

**Comment by rbouqueau** [comment]
@fcsteagu seems related to #305 , correct?

**Comment by fcsteagu** [comment]
yes, indeed.
I only encounter this on Android 5.1.1 (LG Flex 2). Unfortunately I must leave the code choose which library should be used for the decoding. Manually setting the parameter in the config file is not anymore an option..
If you need to test it on the device, I left one to Viet so there should be somewhere in Telecom ParisTech :)
Please let me know if you need some more information related to this subject.

**Comment by rbouqueau** [comment]
Sure, thanks. I close the issue as it is a duplicate of #305. Please re-open if you have news.
